include/ozo/impl/result.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ozo/error.h>
+#include <ozo/type_traits.h>
+#include <ozo/concept.h>
+#include <ozo/impl/row.h>
+
+namespace ozo::impl {
+
+template <typename Rows, typename OidMap, typename Iterator>
+Require<Iterable<Rows> && (ForwardIterator<Iterator> || InsertIterator<Iterator>), error_code>
+convert_result(Rows&& rows, const OidMap& oid_map, Iterator result) {
+    error_code ec = error::ok;
```

Comment:
A right variable indicating no error will be:

```c++
error_code ec;
```

Since ```error::ok``` is still an error code. It is just reserved value to not be confused with zero value error code.

Reply:
done